aws_advanced_python_wrapper/limitless_connection_plugin.py

```
@@ -0,0 +1,543 @@
+                self._synchronously_get_limitless_routers_with_retry(context)
+
+                if (context.get_limitless_routers() is None
+                        or len(context.get_limitless_routers()) == 0
+                        or not context.is_any_routers_available()):
+                    logger.debug("LimitlessRouterServiceImpl.NoRoutersAvailableForRetry")
+
+                    if context.get_connection() is not None:
+                        return
+                    else:
+                        try:
+                            context.set_connection(context.get_connect_func()())
+                            return
+                        except Exception as e:
+                            if self._is_login_exception(e):
+                                raise e
+
+                            raise AwsWrapperError(Messages.get_formatted("LimitlessRouterService.UnableToConnectNoRoutersAvailable"))
+
+            try:
+                selected_host_info = self._plugin_service.get_host_info_from_input_by_strategy(
+                    context.get_limitless_routers(), HostRole.WRITER, "weighted_random")
+                logger.debug("LimitlessRouterServiceImpl.SelectedHostForRetry",
+                             "None" if selected_host_info is None else selected_host_info.host)
+                if selected_host_info is None:
+                    continue
+
+            except UnsupportedOperationError as e:
+                logger.error("LimitlessRouterServiceImpl.IncorrectConfiguration")
+                raise e
+            except AwsWrapperError:
+                continue
+
+            try:
+                context.set_connection(self._plugin_service.connect(selected_host_info, context.get_props()))
+                if context.get_connection() is not None:
+                    return
+
+            except Exception as e:
+                if self._is_login_exception(e):
+                    return
```

In JDBC we raise the exception instead of returning
```suggestion
                    raise e
```